### PR TITLE
wip: format numbers with SI notation

### PR DESCRIPTION
**Pull request by stereosteve**
@danielberkompas first week with Elixir so may be lacking on style points a bit - feedback welcome!

Looking at the [rails version](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/number_helper/number_to_human_size_converter.rb#L11) they are depreciating the `:prefix` / `:si` option, and really that implementation is specifically for bytes.  They don't have an equivalent general purpose SI formatting option, so the parameter is a bit misleading.

So, I was thinking about moving to a separate `si.ex` file that is not beholden to any Rails API.  It would be easy to implement `number_to_human_size` using the SI function.

Let me know what you think.

### CODE REVIEW

**Comment by danielberkompas** [comment]
Looks pretty good! I'll review it in more detail later today.

**Comment by stereosteve** [comment]
I went ahead and moved to separate file and renamed to `Number.SI.format_number`, though `Number.SI.number_to_si` might be a better choice.

Also did some cleanup and added option to trim trailing zeros.

File: lib/number/si.ex [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+  * `:separator` - The string to place between the scaled number and the
+                   prefix + unit. Perhaps you want a space here. Default: ""
+
+  * `:unit` - The unit of measurement, e.g. "M" for Meters. Default: ""
+
+  * `:precision` - The number of decimal places to include. Default: 2
+
+  * `:trim` - Trim trailing zeros. Default: false
+
```
Comment (danielberkompas) [inline lib/number/si.ex]: I'd like to see a section here on how to configure defaults for these options. Like [here](https://github.com/danielberkompas/number/blob/564ce20a9788336249f6bff658a83804eabcee79/lib/number/percentage.ex#L30-L38).

**Comment by danielberkompas** [comment]
Looks great! Fix my one comment and I'll merge.

**Comment by stereosteve** [comment]
Sounds good will do tomorrow. I can also consolidate to one commit if you
prefer.

On Thursday, October 29, 2015, Daniel Berkompas <notifications@github.com>
wrote:

> Looks great! Fix my one comment and I'll merge.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/danielberkompas/number/pull/8#issuecomment-152367111>.
>


-- 
Steve

**Comment by danielberkompas** [comment]
One commit would be nice, thanks!

**Comment by stereosteve** [comment]
Sorry for de-lay... I shortened to `SI.format`, combined to one commit, and opened a new PR.

Closing this in favor of: https://github.com/danielberkompas/number/pull/9

**Pull request closed without merging**